src/tokenize.c

```diff
@@ -69,12 +71,19 @@ static char *DefaultNormalize(char *s, char *dst, size_t *len) {
 
   *len = dstLen;
 
-  size_t newLen = unicode_tolower(dst, dstLen);
-  if (newLen) {
+  size_t newLen = 0;
+  char *longer_dst = unicode_tolower(dst, dstLen, &newLen);
+  if (dstLen != newLen) {
```

Comment:
If newLen is always updated can always do `*len = newLen` (once)

Reply:
Since now we are always updating the length, I changed the unicode_tolower to receive only two args:
`static char* unicode_tolower(char *encoded, size_t *inout_len)`